builtin/providers/aws/resource_aws_elb.go

```diff
@@ -340,7 +340,11 @@ func resourceAwsElbRead(d *schema.ResourceData, meta interface{}) error {
 	d.Set("listener", flattenListeners(lb.ListenerDescriptions))
 	d.Set("security_groups", flattenStringList(lb.SecurityGroups))
 	if lb.SourceSecurityGroup != nil {
-		d.Set("source_security_group", lb.SourceSecurityGroup.GroupName)
+		group := lb.SourceSecurityGroup.GroupName
+		if lb.SourceSecurityGroup.OwnerAlias != nil && *lb.SourceSecurityGroup.OwnerAlias != "" {
+			group = aws.String(*lb.SourceSecurityGroup.OwnerAlias + "/" + *lb.SourceSecurityGroup.GroupName)
+		}
+		d.Set("source_security_group", group)
```

Comment:
It's somewhat related.... when using the ELB's source security group _name_, which you would only use in EC2 Classic, we need the full `amazon-elb/amazon-elb-sg` name to use as a reference inside another security group:

```
    security_groups = [
      "${aws_elb.bar.source_security_group}"
    ]
```

Without this, we have just `amazon-elb-sg` which the API will not find. 

Inside a VPC, we get a default ELB security group in the format of `default_somecrypitic_id`, with the ID of that group stored in `source_security_group_id`. Because we use `_id`'s in VPCs, this should be fine .